src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1075,11 +1075,10 @@ private static NamedTypeSymbol GetDelegateType(BoundExpression expr)
                 methods = constructedMethods.ToImmutableAndFree();
             }
 
-            if (methods.Length == 1 && !extensionMethodsOfSameViabilityAreAvailable)
+            if (methods.Length == 1 && methods[0].Arity == 0 && !extensionMethodsOfSameViabilityAreAvailable)
```

Comment:
When the arity is zero, that means there the method has no type parameters, and since there are no type parameters there is no need for type inference and we can bind to the method's parameter types directly. When there are type parameters, however, this code previously caused a lambda argument to be bound against method parameters types involving type parameters that are not in scope. That was the bug that resulted in the previous incorrect behavior.

Type inference and binding of the lambda against method parameters occurred earlier in the pipeline when the method was considered as a possible candidate. The change here simply prevents binding against an inappropriate delegate type, which is what resulted in a bad lambda binding for error recovery.

---
In reply to: [67019197](https://github.com/dotnet/roslyn/pull/11981#discussion_r67019197) [](ancestors = 67019197)